### ISSUE TITLE
Disabling `showIntroPage` does not work with Chromium

### ISSUE THREAD

**Issue by olifre** [EFForg/privacybadger#2832]
I am trying to disable `showIntroPage` on a Debian 11 system, using the web extension shipped by Debian 11, i.e.:
* Chromium 98.0.4758.80-1~deb11u1
* webext-privacy-badger, version: 2020.10.7-1

Following the recipe at:
https://github.com/EFForg/privacybadger/blob/master/doc/admin-deployment.md
I have created a file `/etc/chromium/policies/managed/managed_storage.json`:
```json
{
  "3rdparty": {
    "extensions": {
      "ooihdgkfebnikkngckonldkbfkdndegf": {
        "showIntroPage": false
      }
    }
  }
}
```
I confirmed this ID matches with what is shown in `chrome://extensions` for Privacy Badger when enabling the developer mode switch. 

After applying this change and purging my profile (via `rm -rf .config/chromium`), I still see the onboarding screen on first start. 

I can confirm the policy shows up correctly in `chrome://policy//`, i.e. it is correctly mapped to Privacy Badger and has the "false" value as expected. 

The following screenshot illusttrates the issue — the left window was opened on first start, in the same session, I opened the right window, and opened `chrome://policy/`. Other policy settings (not for extensions) and the start page are pixelated. 
![badger_policy](https://user-images.githubusercontent.com/166759/154532578-cd9ee061-60e7-4ea0-b5e9-ac333284753f.png)

Am I doing something wrong? Or am I hitting a bug in either Privacy Badger, or Chromium? 

The recipe described in the docs worked just fine with Firefox.

**Comment by ghostwords** [comment]
Hello and thanks for the thorough report!

I think we had a previous report of managed policy settings not being applied in Chrome/Edge (#2770). I don't think we ever got to the root of it.

Privacy Badger ingests managed policy during startup. Does this explain what's happening? Could you talk about your use case a bit more? Perhaps Chromium (sometimes?) finishes reading managed policy settings after Privacy Badger is done initializing.

Also, Privacy Badger version 2020.10.7 is quite out of date at this point. It's (probably) not the cause of this issue, but it's still a problem. You wouldn't use a browser that's this much out of date, right? I know this is something that's up to the Debian package maintainers. I brought this up with them before, but it doesn't seem like they listened. I highly suggested using the latest available version from https://privacybadger.org/ instead.

**Comment by ghostwords** [comment]
OK, I see, thank you.

>Privacy Badger is not active when the browser launches the first time

That's unexpected, and may be related to "unpacked" Privacy Badger not having managed storage upon first launch. This seems like something that any force-installed extension from Chrome Web Store should run into. Is there a race condition between the browser launching and the extension being downloaded? Do other people run into this issue with other extensions? uBlock Origin should be a good candidate to try to reproduce this issue with.

>maybe Chrome/ium should offer an API to query when the policies have finished loading, so Privacy Badger can wait for that for things which need to be known during extension startup?

We could add a managed storage listener to apply the latest managed storage values whenever managed storage updates. (This will also eliminate the requirement that the browser (or Privacy Badger) needs to be restarted before seeing policy updates, which is known to trip up users.)

We could also delay showing the welcome page for a couple of seconds, which should be enough time to work around managed storage taking time to initialize. You could help confirm this approach by force-installing an unpacked debug version of Privacy Badger from GitHub.

How does this sound?

**Comment by olifre** [comment]
> Is there a race condition between the browser launching and the extension being downloaded? Do other people run into this issue with other extensions? uBlock Origin should be a good candidate to try to reproduce this issue with.

Indeed, that seems to be the case. The issue you linked in your latest comment indicates that this seems to be a browser bug indeed. 
Sadly, while the linked bug report did mention that there is a Chromium bug open, that was not linked, and I did not managed to find it yet. 



> Their workaround was to [force an extension restart](https://github.com/gorhill/uBlock/commit/32bd47f05368557044dd3441dcaa414b7b009b39) with `browser.runtime.reload()` on first run on Chromium-based browsers.

This seems like a viable approach until Chromium is (hopefully) fixed at some point. In case there is something I can test, feel free to ping me here. 

And of course, many thanks for delving into this mess!

**Comment by olifre** [comment]
Wow, that does indeed look like a complex startup procedure in Chromium. I wonder how likely it would be to get this fixed in Chromium, which would of course be preferable over adding workarounds such as delayed data access to all extensions.

**Comment by ghostwords** [comment]
I put up a workaround in #2834 where we repeatedly check for managed storage on first run until we either get a non-empty result or we run out of attempts. Let me know if you have any thoughts there please.

**Comment by olifre** [comment]
I think that looks good ­— waiting and polling, and proceeding when managed storage becomes accessible or a timeout is exceeded sounds like the only reasonable approach you can do on the add-on side. :+1:

**Comment by ghostwords** [comment]
Hi @olifre, our workaround is finally live in Privacy Badger version 2022.9.27. Thanks again for helping resolve this issue!

**Comment by olifre** [comment]
Hi @ghostwords , great news and many thanks, that workaround is really appreciated! :+1:
